orchagent/intfsorch.cpp

```diff
@@ -739,6 +796,21 @@ bool IntfsOrch::addRouterIntfs(sai_object_id_t vrf_id, Port &port)
     attr.value.u32 = port.m_mtu;
     attrs.push_back(attr);
 
+    if (isNatSupported)
+    {
+        attr.id = SAI_ROUTER_INTERFACE_ATTR_NAT_ZONE_ID;
+        if (m_nat_zone.find(port.m_alias) == m_nat_zone.end())
+        {
+            attr.value.u32 = DEFAULT_NAT_ZONE_ID;
+        }
+        else
+        {
+            attr.value.u32 = m_nat_zone[port.m_alias];
+        }
```

Comment:
You can revisit the check as if natzone is present in configuration, set the attribute.

Reply:
yes, modified the code to set the attribute if it is present in cache.